### ISSUE TITLE
outdated-version-block (UP036): Flag assignments (or any outdated `sys.version`)

### ISSUE THREAD

**Issue by Avasam** [astral-sh/ruff#16487]
### Summary

I found the following code snippet in `pypa/distutils`, and I'd like [outdated-version-block (UP036)](https://docs.astral.sh/ruff/rules/outdated-version-block/#outdated-version-block-up036) (or a similar rule) to flag usage of `sys.version_info < (3, 8)`:

```py
def _pypy_hack(name):
    PY37 = sys.version_info < (3, 8)
    old_pypy = hasattr(sys, 'pypy_version_info') and PY37
    prefix = not name.endswith(('_user', '_home'))
    pypy_name = 'pypy' + '_nt' * (os.name == 'nt')
    return pypy_name if old_pypy and prefix else name
```

When targetting python 3.9+, in any context this is the same as just writing `False`. Whilst Ruff can't necessarily autofix it *correctly*, it should be flagged for a human to update / remove the legacy code.

**Comment by MichaReiser** [comment]
This does make sense to me, although I'm not sure if it makes sense to include it into UP036 or if we have to rename the rule to `unnecessary-version-check` because it is now no longer about just flagging blocks. 


Related to: https://github.com/astral-sh/ruff/issues/12093

**Comment by gamer191** [comment]
If this was implemented, could it also include other checks that imply outdated versions, such as `os.name == 'ce'` ([if target_version is equal or newer than 3.6](https://github.com/python/cpython/commit/10108a7b9affa61719a1dc1863edb2bdb3402fd1)) and checks that ensure `OPENSSL_VERSION_INFO` is equal or newer than 1.1.1 ([if target_version is equal or newer than 3.10](https://peps.python.org/pep-0644/))

(I'm not 100% confident about the OpenSSL one)

**Comment by Avasam** [comment]
@gamer191 I'm not against such checks. But I'm pretty sure that should be its own, dedicated "old code migration"-kind of rule and new feature request.
